sechub-commons-model/src/test/java/com/mercedesbenz/sechub/commons/model/SecHubRemoteCredentialSupportTest.java

```diff
@@ -0,0 +1,181 @@
+        assertNotNull(container2c);
+        assertEquals("userX", container2c.getUser());
+
+        assertTrue(credentials3.isEmpty());
+    }
+
+    @Test
+    void match_remote_credentials_pattern_by_type() {
+        /* prepare */
+        String json = TestFileReader.loadTextFile(new File("./src/test/resources/sechub_remote_credentials_complex_config.json"));
+        SecHubRemoteCredentialConfiguration configuration = supportToTest.getRemoteCredentialConfigurationFromJSONString(json);
+        String type1 = "docker";
+        String type2 = "git";
+        String type3 = "unknown";
+        String location = "https://example.url.com";
+
+        /* execute */
+        List<SecHubRemoteCredentialContainer> credentials1 = supportToTest.resolveCredentialsForLocation(configuration, location, type1);
+        List<SecHubRemoteCredentialContainer> credentials2 = supportToTest.resolveCredentialsForLocation(configuration, location, type2);
+        List<SecHubRemoteCredentialContainer> credentials3 = supportToTest.resolveCredentialsForLocation(configuration, location, type3);
+
+        /* test */
+        assertEquals(1, credentials1.size());
+        assertEquals(2, credentials2.size());
+
+        SecHubRemoteCredentialContainer container1 = credentials1.get(0);
+        assertNotNull(container1);
+        assertEquals("user3", container1.getUser());
+        assertEquals("token3", container1.getPassword());
+
+        SecHubRemoteCredentialContainer container2 = credentials2.get(0);
+        assertNotNull(container2);
+        assertEquals("user2", container2.getUser());
+
+        SecHubRemoteCredentialContainer container2b = credentials2.get(1);
+        assertNotNull(container2b);
+        assertEquals("user3", container2b.getUser());
+
+        assertTrue(credentials3.isEmpty());
+    }
+
+    @ParameterizedTest
+    @EmptySource
+    @NullSource
+    void match_remote_credential_pattern_by_type_with_empty_or_null_type(String string) {
+        /* prepare */
+        String json = TestFileReader.loadTextFile(new File("./src/test/resources/sechub_remote_credentials_complex_config.json"));
+        SecHubRemoteCredentialConfiguration configuration = supportToTest.getRemoteCredentialConfigurationFromJSONString(json);
+
+        /* execute */
+        List<SecHubRemoteCredentialContainer> credentials = supportToTest.resolveCredentialsForLocation(configuration, string);
+
+        /* test */
+        assertTrue(credentials.isEmpty());
+    }
+
+    @ParameterizedTest
+    @EmptySource
+    @NullSource
+    void match_remote_credential_pattern_by_location_with_empty_or_null_location(String string) {
+        /* prepare */
+        String json = TestFileReader.loadTextFile(new File("./src/test/resources/sechub_remote_credentials_complex_config.json"));
+        SecHubRemoteCredentialConfiguration configuration = supportToTest.getRemoteCredentialConfigurationFromJSONString(json);
```

Comment:
```suggestion
        SecHubRemoteCredentialConfiguration configuration = SecHubRemoteCredentialConfiguration.fromJSONString(json); 
```
A unit test should not need to call the object which is tested to prepare something.